product/views.py

```diff
@@ -1,3 +1,63 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.views import View
+from django.http import JsonResponse
+from django.db.models import Prefetch
+from user.models import Creator,User
+from request.models import CourseRequest
+from product.models import (
+    Course,
+    Category,
+    SubCategory,
+    ProductImage,
+    Product,
+    ProductLike
+)
+
+#클래스 좋아요 기능
+class CourseLikeView(View):
+   # @id_auth
+    def post(self, request, product_pk):
+        try:
+           # user = request.user
+            user   = User.objects.get(id = 1)
+            course = Product.objects.get(id = product_pk)
+
+            like, course_like = ProductLike.objects.get_or_create(user = user, product = course)
+            if not course_like:
+                return JsonResponse({'message':'ALREADY_LIKE'},status = 400)
+            return JsonResponse({'message':'SUCCESS'}, status = 201)
+        except ProductLike.Doesnotexist:
+            return JsonResponse({'message':'INVALID_ACCESS'}, status = 404)
+#좋아요 취소
+   # @id_auth
+    def delete(self, request, product_pk):
+        #user = request.user
+        user        = User.objects.get(id = 1)
+        course      = Product.objects.get(id = product_pk)
+        course_like = ProductLike.objects.get(user = user, product = course)
+        course_like.delete()
+        return JsonResponse({'message':'SUCCESS'}, status = 204)
+
+#크리에이터 메인화면
+class CourseListView(View):
+    def get(self, request, category_pk):
+        category    = Category.objects.get(id = category_pk)
+        subcategory = SubCategory.objects.filter(category = category)
+        products    = Product.objects.prefetch_related('course_set','course_set__courserequest_set').select_related('sub_category','creator', 'creator__user').filter(sub_category__in = subcategory)
+
+        text = [{
+            'id'           : product.id,
+            'sub_category' : product.sub_category.name,
+            'alt'          : product.name,
+            'cover_image'  : product.course_set.get().courserequest_set.all()[0].thumbnail,
+          
+            'creator'      : product.creator.user.nickname,
+            'name'         : product.name,
+            'price'        : product.price,
+            'like'         : ProductLike.objects.filter(product_id = product.id).count()
```

Comment:
여기도 이미 caching해서 가져온 값을 사용하는게 좋을 것 같습니다. 
```suggestion
            'like'         : product..
```